File: cpp/src/sensor.cpp


#include "sensor.h"
#include "iostream"

#include <cstdlib>

Sensor::Sensor()
{
    srand(time(NULL));
}

double Sensor::popNextPressurePsiValue()
{
    // placeholder implementation that simulates a real sensor in a real tire
    float offset = (float) (rand() % 8);
    //std::cout << "offset: " << offset << "\n";
    double pressure = 15 + offset;
    return pressure;
}

File: typescript/src/Sensor.ts
const offset = 16;

export default class Sensor {
  public popNextPressurePsiValue(): number {
    return offset + this.samplePressure();
  }

  private samplePressure(): number {
    // placeholder implementation that simulates a real sensor in a real tire
    return 6 * Math.random() * Math.random();
  }
}
